test/algorithm/adjacent_remove_if.cpp

```diff
@@ -0,0 +1,199 @@
+    ia[0].reset(new int(0));
+    ia[1].reset(new int(1));
+    ia[2].reset(new int(2));
+    ia[3].reset(new int(3));
+    ia[4].reset(new int(4));
+    ia[5].reset(new int(2));
+    ia[6].reset(new int(3));
+    ia[7].reset(new int(4));
+    ia[8].reset(new int(2));
+    Iter r = ranges::adjacent_remove_if(::as_lvalue(ranges::make_iterator_range(Iter(ia), Sent(ia+sa))), pred());
+    CHECK(base(r) == ia + sa-2);
+    CHECK(*ia[0] == 0);
+    CHECK(*ia[1] == 1);
+    CHECK(*ia[2] == 3);
+    CHECK(*ia[3] == 4);
+    CHECK(*ia[4] == 3);
+    CHECK(*ia[5] == 4);
+    CHECK(*ia[6] == 2);
+}
+
+struct S
+{
+    int i;
+};
+
+int main()
+{
+    test_iter<forward_iterator<int*> >();
+    test_iter<bidirectional_iterator<int*> >();
+    test_iter<random_access_iterator<int*> >();
+    test_iter<int*>();
+    test_iter<forward_iterator<int*>, sentinel<int*>>();
+    test_iter<bidirectional_iterator<int*>, sentinel<int*>>();
+    test_iter<random_access_iterator<int*>, sentinel<int*>>();
+    test_range<forward_iterator<int*> >();
+    test_range<bidirectional_iterator<int*> >();
+    test_range<random_access_iterator<int*> >();
+    test_range<int*>();
+    test_range<forward_iterator<int*>, sentinel<int*>>();
+    test_range<bidirectional_iterator<int*>, sentinel<int*>>();
+    test_range<random_access_iterator<int*>, sentinel<int*>>();
+
+    test_iter_rvalue<forward_iterator<std::unique_ptr<int>*> >();
+    test_iter_rvalue<bidirectional_iterator<std::unique_ptr<int>*> >();
+    test_iter_rvalue<random_access_iterator<std::unique_ptr<int>*> >();
+    test_iter_rvalue<std::unique_ptr<int>*>();
+    test_iter_rvalue<forward_iterator<std::unique_ptr<int>*>, sentinel<std::unique_ptr<int>*>>();
+    test_iter_rvalue<bidirectional_iterator<std::unique_ptr<int>*>, sentinel<std::unique_ptr<int>*>>();
+    test_iter_rvalue<random_access_iterator<std::unique_ptr<int>*>, sentinel<std::unique_ptr<int>*>>();
+
+    test_range_rvalue<forward_iterator<std::unique_ptr<int>*> >();
+    test_range_rvalue<bidirectional_iterator<std::unique_ptr<int>*> >();
+    test_range_rvalue<random_access_iterator<std::unique_ptr<int>*> >();
+    test_range_rvalue<std::unique_ptr<int>*>();
+    test_range_rvalue<forward_iterator<std::unique_ptr<int>*>, sentinel<std::unique_ptr<int>*>>();
+    test_range_rvalue<bidirectional_iterator<std::unique_ptr<int>*>, sentinel<std::unique_ptr<int>*>>();
+    test_range_rvalue<random_access_iterator<std::unique_ptr<int>*>, sentinel<std::unique_ptr<int>*>>();
+
+    {
+        // Check projection
+        S ia[] = {S{0}, S{1}, S{1}, S{1}, S{4}, S{2}, S{2}, S{4}, S{2}};
+        constexpr auto sa = ranges::size(ia);
+        S* r = ranges::adjacent_remove_if(ia, ranges::equal_to{}, &S::i);
+        CHECK(r == ia + sa-3);
+        CHECK(ia[0].i == 0);
+        CHECK(ia[1].i == 1);
+        CHECK(ia[2].i == 4);
+        CHECK(ia[3].i == 2);
+        CHECK(ia[4].i == 4);
+        CHECK(ia[5].i == 2);
+    }
+
+#if 0 // Should this even be included?
```

Comment:
Why not include it? (It certainly shouldn't be present-but-disabled.)

Reply:
> Why not include it?

I wasn't sure if range-v3 would be dropping support for rvalues like C++20.

> (It certainly shouldn't be present-but-disabled.)

Disabled to bring your attention to it (I guess I could have made a review comment).